lib/ui/painting/image_generator_registry.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <algorithm>
+
+#include "flutter/lib/ui/painting/image_generator_registry.h"
+#include "third_party/skia/include/codec/SkCodec.h"
+#include "third_party/skia/include/core/SkImageGenerator.h"
+#include "third_party/skia/src/codec/SkCodecImageGenerator.h"
+#ifdef OS_MACOSX
+#include "third_party/skia/include/ports/SkImageGeneratorCG.h"
+#elif OS_WIN
+#include "third_party/skia/include/ports/SkImageGeneratorWIC.h"
+#endif
+
+namespace flutter {
+
+ImageGeneratorRegistry::ImageGeneratorRegistry() : weak_factory_(this) {
+  AddFactory(
+      [](sk_sp<SkData> buffer) {
+        return BuiltinSkiaCodecImageGenerator::MakeFromData(buffer);
+      },
+      0);
+
+#ifdef OS_MACOSX
+  AddFactory(
+      [](sk_sp<SkData> buffer) {
+        auto generator = SkImageGeneratorCG::MakeFromEncodedCG(buffer);
+        return BuiltinSkiaImageGenerator::MakeFromGenerator(
+            std::move(generator));
+      },
+      0);
+#elif OS_WIN
+  add(
+      [](sk_sp<SkData> buffer) {
+        auto generator = SkImageGeneratorWIC::MakeFromEncodedWIC(buffer);
+        return BuiltinSkiaImageGenerator::makeFromGenerator(
+            std::move(generator));
+      },
+      0);
+#endif
```

Comment:
Since we're creating API surface for this, wouldn't we want to move this to the specific embeddings?

I think at the very least we'd want a TODO here linking to an issue to do this when it's exposed through the embedder API

Reply:
Yup for sure, I added a todo & issue for this. The embedder API will be extended for this.